python/ray/autoscaler/updater.py

```diff
@@ -294,10 +294,16 @@ def do_update(self):
         node_tags = self.provider.node_tags(self.node_id)
         logger.debug("Node tags: {}".format(str(node_tags)))
 
+        if node_tags.get(TAG_RAY_RUNTIME_CONFIG) == self.runtime_hash:
```

Comment:
Won't this always run `run_init` even when it has already been run?

Reply:
Yeah, in the case where we stop (not **terminate**) and then restart a node. This doesn't solve the problem of rerunning `setup_commands` (but those _should_ be minimal when Docker is in use).